nvflare/fuel/f3/cellnet/core_cell.py

```diff
@@ -1159,7 +1159,18 @@ def _send_target_messages(
             if ep:
                 reachable_targets[t] = ep
             else:
-                self.log_error(f"cannot send to '{t}': {err}", tm.message)
+                msg = Message(headers=copy.copy(tm.message.headers), payload=tm.message.payload)
+                msg.add_headers(
+                    {
+                        MessageHeaderKey.CHANNEL: tm.channel,
+                        MessageHeaderKey.TOPIC: tm.topic,
+                        MessageHeaderKey.FROM_CELL: self.my_info.fqcn,
+                        MessageHeaderKey.TO_CELL: t,
+                        MessageHeaderKey.ORIGIN: self.my_info.fqcn,
+                        MessageHeaderKey.DESTINATION: t,
+                    }
+                )
```

Comment:
You are adding these headers later for the reachable_targets as well. Should you change to add these headers to all the target_msgs regardless? That will also improve the readability.

Reply:
As a good practice, we should not change the input argument. So we always create a new instance of Message and copy the headers to keep the target_msgs intact.

Reply:
I mean this is the same logic as the target_msgs to create a Message and add headers to it, not to change the input argument.